flink-core/src/test/java/org/apache/flink/core/fs/SafetyNetCloseableRegistryTest.java

```diff
@@ -27,18 +27,31 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
 import java.io.Closeable;
 import java.io.IOException;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.doThrow;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
 /**
  * Tests for the {@link SafetyNetCloseableRegistry}.
  */
+@RunWith(PowerMockRunner.class)
```

Comment:
Mockito is not recommended for Flink tests.
An alternative is to introduce a new constructor `SafetyNetCloseableRegistry(CloseableReaperThread)` and a test class which overrides `CloseableReaperThread`.

Reply:
Thanks for reivew, @zhuzhurk . I will considier your comment. But `CloseableReaperThread ` is final class and can't be inherited, I will change this for override.